tapdb/assets_store.go

```diff
@@ -1404,13 +1402,14 @@ func (a *AssetStore) insertAssetWitnesses(ctx context.Context,
 			copy(splitCommitmentProof, b.Bytes())
 		}
 
-		err = db.InsertAssetWitness(ctx, PrevInput{
+		err = db.UpsertAssetWitness(ctx, PrevInput{
 			AssetID:              assetID,
 			PrevOutPoint:         prevOutpoint,
 			PrevAssetID:          prevID.ID[:],
 			PrevScriptKey:        prevID.ScriptKey.CopyBytes(),
 			WitnessStack:         witnessStack,
 			SplitCommitmentProof: splitCommitmentProof,
+			WitnessIndex:         int32(idx),
```

Comment:
Ah so we were missing this all along? Or we were relying on implicit read order here to keep the expected serialization the same?

Reply:
Or seems the assumption is that no multi-input spends exist today in the DB?

Reply:
Yes, I think we were lucky and the implicit read order was correct so far. And we only have one itest that actually does use multiple inputs.